File: demo-service-spring-impl/src/main/java/com/github/zubmike/service/demo/types/Zone.java
package com.github.zubmike.service.demo.types;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "zones")
@Cache(region = "zone", usage = CacheConcurrencyStrategy.READ_WRITE)
public class Zone extends BasicEntityDictItem {

	private static final long serialVersionUID = 1L;

	@Column(name = "max_size")
	private int maxSize;

	public int getMaxSize() {
		return maxSize;
	}

	public void setMaxSize(int maxSize) {
		this.maxSize = maxSize;
	}
}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/types/Zone.kt
package com.github.zubmike.service.demo.types

import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import com.github.zubmike.service.types.BasicEntityDictItem

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "zones")
@Cache(region = "zone", usage = CacheConcurrencyStrategy.READ_WRITE)
class Zone : BasicEntityDictItem() {

    @Column(name = "max_size")
    var maxSize: Int = 0

}
